eng/intellisense.targets

```diff
@@ -0,0 +1,32 @@
+<Project>
+
+  <PropertyGroup>
+      <!-- Search for the documentation file in the intellisense package and otherwise pick up the generated one. -->
+    <IntellisensePackageFolderName Condition="'$(IsPackable)' == '' or '$(IsPackable)' == 'false'">net</IntellisensePackageFolderName>
+    <IntellisensePackageFolderName Condition="'$(IsPackable)' == 'true'">dotnet-plat-ext</IntellisensePackageFolderName>
```

Comment:
Instead of having a hardcoded`'net'` value in `Directory.Build.props`, we decide here the value we want.

There are other folders in the intellisense nupkg: `netstandard` and `windowsdesktop`. I am unsure if we need to add more conditions to also include those.

I suspect the `windowsdesktop` folder would have to be used in the winforms repo infrastructure, not here.

Reply:
Yeah this is the only part I'm unsure about. Maybe we'll need those other folders as well.

Reply:
You shouldn't need netstandard as it only contains the netstandard folder as it only contains the netstandard.xml which we don't expose in our targeting pack. It was only used by the netstandard repo, which is now archived.

Neither do I think that you need windowsdesktop.

But as Eric said in the stand-up on Monday, we shouldn't condition on `IsPackable` but probe in a priority order instead:
1. Probe inside the "net" folder
2. If the file doesn't exist, probe the "dotnet-plat-ext" folder.
3. If the file still doesn't exist, use the compiler generated xml file.

But we should only do that when `$(UseIntellisenseDocumentationFile)` isn't true as that probing would otherwise be a waste of time.

Reply:
This should be addressed.